### PR TITLE
fix(agw): Gracefully handle died process

### LINKED ISSUE #11388: Died process is handled gracefully in metrics gathering
In the `_collect_service_metrics` function, we determine a service's PID, and then query that PID using psutil. That causes a race condition: When the process dies after the PID has been queried, we get a `psutil.NoSuchProcess` exception. Instead of failing with an unhandled exception, log a warning and continue with the next service.

Sentry Issue: [LAB-AGWS-PYTHON-6N](https://sentry.io/organizations/lf-9c/issues/2979646908/?referrer=github_integration)

```
AttributeError: _cache
  File "psutil/_common.py", line 447, in wrapper
    ret = self._cache[fun]

FileNotFoundError: [Errno 2] No such file or directory: '/proc/508688/stat'
  File "psutil/_pslinux.py", line 1576, in wrapper
    return fun(self, *args, **kwargs)
  File "psutil/_common.py", line 450, in wrapper
    return fun(self)
  File "psutil/_pslinux.py", line 1618, in _parse_stat_file
    with open_binary("%s/%s/stat" % (self._procfs_path, self.pid)) as f:
  File "psutil/_common.py", line 711, in open_binary
    return open(fname, "rb", **kwargs)

NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=508688)
(2 additional frame(s) were not displayed)
...
  File "/usr/local/lib/python3.8/dist-packages/magma/magmad/metrics.py", line 291, in _collect_service_metrics
    cpu_percentage = p.cpu_percent(interval=1)
  File "__init__.py", line 990, in cpu_percent
    pt2 = self._proc.cpu_times()
  File "psutil/_pslinux.py", line 1576, in wrapper
    return fun(self, *args, **kwargs)
  File "psutil/_pslinux.py", line 1769, in cpu_times
    values = self._parse_stat_file()
  File "psutil/_pslinux.py", line 1583, in wrapper
    raise NoSuchProcess(self.pid, self._name)

Task exception was never retrieved
future: <Task finished name='Task-9' coro=<metrics_collection_loop() done, defined at /usr/local/lib/python3.8/dist-packages/magma/magmad/metrics.py:139> exception=psutil.NoSuchProcess process no longer exists (pid=508688)>
```

### PR DESCRIPTION
**Pull request by jheidbrink**
fixes #11388

## Summary

Handle died process in metrics collection gracefully by logging a warning and continuing with the next service

## Test Plan

untested

### CODE REVIEW

**Review by themarwhal — APPROVED** [review]
nice :)

File: orc8r/gateway/python/magma/magmad/metrics.py [hunk 1]
```diff
@@ -287,13 +287,18 @@ def _collect_service_metrics():
...
-                    service_name=service,
-                ), cpu_percentage,
-            )
+            try:
+                p = psutil.Process(pid=pid)
+                cpu_percentage = p.cpu_percent(interval=1)
+            except psutil.NoSuchProcess:
+                logging.warning("When collecting CPU usage for service %s: Process with PID %d no longer exists.", service, pid)
+                continue
+            else:
```
Comment (hcgatewood) [inline orc8r/gateway/python/magma/magmad/metrics.py:296]: This indentation seems a little off... what's this "else" corresponding with?

Comment (Neudrino) [reply orc8r/gateway/python/magma/magmad/metrics.py:296]: The `else` is aligned correctly with the `try` block here and will only be executed if the `except` statement is not executed.

**Review by Neudrino — APPROVED** [review]

File: orc8r/gateway/python/magma/magmad/metrics.py (hunk 1 above)
Comment (hcgatewood) [reply orc8r/gateway/python/magma/magmad/metrics.py:296]: TIL 🙂

**Pull request merged**